consent/admin/__init__.py

```diff
@@ -0,0 +1,101 @@
+# -*- coding: utf-8 -*-
+"""
+Django admin integration for the Consent application.
+"""
+
+from __future__ import absolute_import, unicode_literals
+
+from django.contrib import admin
+from simple_history.admin import SimpleHistoryAdmin
+
+from consent.models import (  # pylint:disable=no-name-in-module
+    DataSharingConsent,
+    HistoricalDataSharingConsent,
+)
+
+
+class HistoricalDataSharingConsentInlineAdmin(admin.TabularInline):
+    """
+    Inline admin view for ``DataSharingConsent`` history.
+    """
+
+    model = HistoricalDataSharingConsent
+
+    fields = (
+        'granted',
+        'history_date',
+    )
+
+    readonly_fields = (
+        'granted',
+        'history_date',
+    )
+
+    def has_add_permission(self, request):
+        """
+        Disable add permission for HistoricalDataSharingConsent.
+        """
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        """
+        Disable deletion permission for HistoricalDataSharingConsent.
+        """
+        return False
+
+
+@admin.register(DataSharingConsent)
+class DataSharingConsentAdmin(SimpleHistoryAdmin):
+    """
+    Django admin model for PendingEnrollment
+    """
+
+    class Meta(object):
+        """
+        Meta class for ``DataSharingConsentAdmin``.
+        """
+
+        model = DataSharingConsent
+
+    readonly_fields = (
+        'enterprise_customer',
+        'username',
+        'course_id',
+        'granted',
+        'exists',
+    )
+
+    list_display = (
+        'enterprise_customer',
+        'username',
+        'course_id',
+        'granted',
+        'exists',
+    )
+
+    ordering = (
+        "username",
+    )
+
+    search_fields = (
+        'enterprise_customer__name',
+        'enterprise_customer__uuid',
+        'username',
+        'course_id',
+    )
+
+    inlines = (
+        HistoricalDataSharingConsentInlineAdmin,
```

Comment:
There isn't actually a ForeignKey between the DSC and hDSC, so we either need to do some fancy magic here to get the right queryset, or we need to skip this entirely. For now, I recommend the latter. This is actually preventing the migration from working on my devstack.